web/src/reducers/roles.js

```diff
@@ -30,23 +30,23 @@ export default function rolesReducer(state = initialState, action = {}) {
         ...state,
         organisation: {
           ...state.organisation,
-          [action.payload.id]: action.payload.data,
+          [action.payload.id]: {...action.payload.data, ...state.organisation[action.payload.id]},
         },
       };
     case ADD_APPLICATION_ROLES:
       return {
         ...state,
         application: {
           ...state.application,
-          [action.payload.id]: action.payload.data,
+          [action.payload.id]: {...action.payload.data, ...state.application[action.payload.id]},
         },
       };
     case ADD_SPACE_ROLES:
       return {
         ...state,
         space: {
           ...state.space,
-          [action.payload.id]: action.payload.data,
+          [action.payload.id]: {...action.payload.data, ...state.space[action.payload.id]},
```

Comment:
here the action.payload.data properties will be overridden by previous state properties is this what you intended ?

Reply:
action.payload was overwriting some properties of the state which should not be effected, so I want to retain those values of the state, thats why I did that.